phac-nml/irida-next
# Sign in enhancements

## What does this PR do and why?
The email & password login form is now being displayed after clicking the local account button when omni auth providers are specified.
Fixes #190.

## Screenshots or screen recordings
![Screenshot from 2023-10-12 16-13-11](https://github.com/phac-nml/irida-next/assets/325703/f2aa9c74-c7b6-4ffe-9217-2d9581707bab)
![Screenshot from 2023-10-12 16-11-33](https://github.com/phac-nml/irida-next/assets/325703/cff3c741-3e06-4242-82dd-85b450cbc373)
![Screenshot from 2023-10-12 16-12-21](https://github.com/phac-nml/irida-next/assets/325703/7ade95aa-6187-404c-9e78-31a98dae5e84)

## How to set up and validate locally
Test with no omni auth providers:
1. In a terminal run `bin/dev`.
2. In a browser navigate to `http://localhost:3000/.
3. Verify only email & password login is displayed.

Test with omni auth providers:
1. In a terminal run `OMNIAUTH_PROVIDERS="developer,saml,azure_activedirectory_v2" bin/dev`.
2. In a browser navigate to `http://localhost:3000/.
3. Verify the selection page is displayed.
4. Click the `Sign in with Local Account` button.
5. Verify the page refreshes with email & password form login displayed.
6. Click the `Return to sign in selection` button.
7. Verify the selection page is displayed.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
